container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1748,6 +2070,140 @@ func (manager *KubernetesManager) RunExecCommandWithStreamedOutput(
+					"sh",
+					"-c",
+					"sleep 10000000s",
+				},
+				Args:       nil,
+				WorkingDir: "",
+				Ports:      nil,
+				EnvFrom:    nil,
+				Env:        nil,
+				Resources: apiv1.ResourceRequirements{
+					Limits:   nil,
+					Requests: nil,
+					Claims:   nil,
+				},
+				ResizePolicy: nil,
+				VolumeMounts: []apiv1.VolumeMount{
+					{
+						Name:             hostVolumeName,
+						ReadOnly:         false,
+						MountPath:        hostMountPath,
+						SubPath:          "",
+						MountPropagation: nil,
+						SubPathExpr:      "",
+					},
+				},
+				VolumeDevices:            nil,
+				LivenessProbe:            nil,
+				ReadinessProbe:           nil,
+				StartupProbe:             nil,
+				Lifecycle:                nil,
+				TerminationMessagePath:   "",
+				TerminationMessagePolicy: "",
+				ImagePullPolicy:          "",
+				SecurityContext: &apiv1.SecurityContext{
+					Privileged:               &isPrivileged,
+					Capabilities:             nil,
+					SeccompProfile:           nil,
+					ProcMount:                nil,
+					ReadOnlyRootFilesystem:   nil,
+					AllowPrivilegeEscalation: nil,
+					RunAsNonRoot:             nil,
+					RunAsGroup:               nil,
+					RunAsUser:                nil,
+					SELinuxOptions:           nil,
+					WindowsOptions:           nil,
+				},
+				Stdin:     false,
+				StdinOnce: false,
+				TTY:       false,
+			},
+		},
+		[]apiv1.Volume{
+			{
+				Name:         hostVolumeName,
+				VolumeSource: manager.GetVolumeSourceForHostPath("/"), // mount the entire host filesystem in this volume
+			},
+		},
+		"",
+		"",
+		nil,
+		map[string]string{
+			"kubernetes.io/hostname": nodeName,
+		},
+		hasHostPidAccess,
+		hasHostNetworkAccess,
+	)
+	defer func() {
+		// Don't block on removing this remove directory pod because this can take a while sometimes in k8s
+		go func() {
+			removeCtx := context.Background()
+			if removeDataDirPod != nil {
+				err := manager.RemovePod(removeCtx, removeDataDirPod)
+				if err != nil {
+					logrus.Warnf("Attempted to remove pod '%v' in namespace '%v' but an error occurred:\n%v", removeDataDirPod.Name, namespace, err.Error())
+					logrus.Warn("You may have to remove this pod manually.")
+				}
+			}
+		}()
+	}()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating pod '%v' in namespace '%v'.", removeDataDirPodName, namespace)
+	}
+
+	removeDirSuccessExitCode := int32(0)
+	dirPathToRemoveAndEmpty := fmt.Sprintf("%v%v", hostMountPath, dirPathToRemove)
+	removeDirCmd := []string{"rm", "-rf", dirPathToRemoveAndEmpty}
+	output := &bytes.Buffer{}
+	concurrentWriter := concurrent_writer.NewConcurrentWriter(output)
+	resultExitCode, err := manager.RunExecCommand(
+		removeDataDirPod.Namespace,
+		removeDataDirPod.Name,
+		removeContainerName,
+		removeDirCmd,
+		concurrentWriter,
+		concurrentWriter,
+	)
+	logrus.Debugf("Output of clean '%v': %v, exit code: %v", removeDirCmd, output.String(), resultExitCode)
```

Comment:
Looks like this was pulled from the `Clean()` function earlier, but this function is no longer `Clean` (it's just removing a directory)